src/Stack/Setup.hs

```diff
@@ -609,6 +618,33 @@ getGhcBuild menv = do
         $logDebug ("Using " <> T.pack s <> " GHC build")
         return (CompilerBuildSpecialized s)
 
+-- | Encode an OpenBSD version (like "6.1") into a valid argument for
+-- CompilerBuildSpecialized, so "maj6-min1". Later version numbers are prefixed
+-- with "r".
+-- The result r must be such that "ghc-" ++ r is a valid package name,
+-- as recognized by parsePackageNameFromString.
```

Comment:
That is: the naming scheme is constrained by parsePackageNameFromString, which is called at https://github.com/commercialhaskell/stack/blob/356fc5a35c2e87a4d3b9c4180452acd18e0b4b00/src/Stack/Setup.hs#L839 and another location in the same file on the ghc-build + other stuff. Not sure why, but not an accident.
/cc @borsboom